testsuite/MDAnalysisTests/analysis/test_distances.py

```diff
@@ -235,3 +235,136 @@ def test_between_return_type(self, dists, group, ag, ag2):
             dists
         )
         assert isinstance(actual, MDAnalysis.core.groups.AtomGroup)
+
+
+class TestAtomicDistances(object):
+    @staticmethod
+    @pytest.fixture()
+    def ad_u():
+        return MDAnalysis.Universe(GRO, XTC)
+
+    @staticmethod
+    @pytest.fixture()
+    def ad_ag1(ad_u):
+        return ad_u.atoms[100:105]
+
+    @staticmethod
+    @pytest.fixture()
+    def ad_ag2(ad_u):
+        return ad_u.atoms[300:305]
+
+    @staticmethod
+    @pytest.fixture()
+    def ad_ag3(ad_u):
+        # more atoms than expected (exception)
+        return ad_u.atoms[:7]
+
+    @staticmethod
+    @pytest.fixture()
+    def ad_ag4():
+        u = MDAnalysis.Universe(GRO)
+        # different trajectory, no XTC (exception)
+        return u.atoms[-5:]
+
+    @staticmethod
+    @pytest.fixture()
+    def expected_scipy(ad_ag1, ad_ag2):
+        expected = np.zeros((len(ad_ag1.universe.trajectory),
+                             ad_ag1.atoms.n_atoms))
+
+        # calculate distances without PBCs using scipy
+        for i, ts in enumerate(ad_ag1.universe.trajectory):
+            expected[i] = np.diag(scipy.spatial.distance.cdist
+                                  (ad_ag1.positions,
+                                   ad_ag2.positions))
+        return expected
+
+    @staticmethod
+    @pytest.fixture()
+    def expected_dist(ad_ag1, ad_ag2):
+        expected = np.zeros((len(ad_ag1.universe.trajectory),
+                             ad_ag1.atoms.n_atoms))
+
+        # calculate distances without PBCs using dist()
+        for i, ts in enumerate(ad_ag1.universe.trajectory):
+            expected[i] = MDAnalysis.analysis.distances.dist(ad_ag1,
+                                                             ad_ag2)[2]
+        return expected
+
+    @staticmethod
+    @pytest.fixture()
+    def expected_pbc(ad_ag1, ad_ag2):
+        expected = np.zeros((len(ad_ag1.universe.trajectory),
+                             ad_ag1.atoms.n_atoms))
+
+        # calculate distances with PBCs
+        for i, ts in enumerate(ad_ag1.universe.trajectory):
+            # non-pbc x, y, z distances
+            dist = np.abs(ad_ag1.positions - ad_ag2.positions)
+
+            # box size (lx, ly, lz)
+            box = ad_ag1.dimensions[:3]
+
+            # apply minimum image convention i.e. take box - dist
+            # when dist > box / 2
+            dist = np.where(dist > box / 2, box - dist, dist)
+
+            # desired dist = sqrt((x2 - x1)^2 + (y2 - y1)^2 + (z2 - z1)^2)
+            expected[i] = np.sqrt(np.square(dist).sum(axis=1))
```

Comment:
again just use `dist` here.  we're not testing that our distance calculations work here (we do that elsewhere).  We're testing that **this class** is correctly applying those functions.